### PR TITLE
Adding E2E tests for .Net core projects

### PR DESCRIPTION
**Pull request by mishra14**
Fixes: https://github.com/NuGet/Home/issues/4282

This should cover basic E2E scenarios for .Net Core projects.

Note - Still waiting on E2E machine before having these pass completely, but they pass locally. Good to have some eyes on this in the mean time.

//cc: @emgarten @alpaix @jainaashish @rohit21agrawal @nkolev92 @zhili1208

### CODE REVIEW

File: test/EndToEnd/vs.ps1 [hunk 1]
```diff
@@ -30,6 +30,35 @@ function New-BuildIntegratedProj
...
 }
 
+function Wait-OnNetCoreRestoreCompletion{
+     param(
+        [parameter(Mandatory = $true)]
+        $Project
+    )
+    
+    $NetCoreLockFilePath = Get-NetCoreLockFilePath $project   
+    while (!(Test-Path $NetCoreLockFilePath)) { Start-Sleep -Seconds 1 }
```
Comment (alpaix) [inline test/EndToEnd/vs.ps1]: Limit total time of waiting here. Shouldn't be waiting forever.

Comment (mishra14) [reply test/EndToEnd/vs.ps1]: Yes!

File: test/EndToEnd/ProjectTemplates/NetCoreConsoleApp.zip/ConsoleApp14.csproj [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (alpaix) [inline test/EndToEnd/ProjectTemplates/NetCoreConsoleApp.zip/ConsoleApp14.csproj]: Redundant file?

**Comment by alpaix** [comment]
What about tests for ASP.NET project? The key difference the template contains package references that should be restored on project creation.

File: test/EndToEnd/vs.ps1 [hunk 3]
```diff
@@ -30,6 +30,44 @@ function New-BuildIntegratedProj
...
 
+function Wait-OnNetCoreRestoreCompletion{
+     param(
+        [parameter(Mandatory = $true)]
+        $Project
+    )
+    
+    $NetCoreLockFilePath = Get-NetCoreLockFilePath $Project
+    [int]$count = 0
+    while (!(Test-Path $NetCoreLockFilePath))
```
Comment (alpaix) [inline test/EndToEnd/vs.ps1]: Would be better with this:
```ps1
$timeout = New-Timespan -Minutes 2
$sw = [Diagnostics.Stopwatch]::StartNew()
while (!(Test-Path $NetCoreLockFilePath)) {
  if ($sw.elapsed -ge $timeout) {
    break
  }
  Start-Sleep -Seconds 1
}
```

Comment (alpaix) [reply test/EndToEnd/vs.ps1]: Also make sure to fail the test with timeout message!

**Review by emgarten — APPROVED** [review]
LGTM, tests for the ASP projects such as MVC would be nice to stress the product a bit more.

**Comment by mishra14** [comment]
@alpaix I have added net core web app tests. Waiting for CI to run them.
I dont think we need to fail tests if the wait times out since each test asserts the asset file right after project creation.

**Comment by alpaix** [comment]
Timeout is an indication of possible hang. There should be some hint in a log

**Review by alpaix — APPROVED** [review]
Looks ✨

**Comment by emgarten** [comment]
:shipit:

**Pull request merged**